apps/theme/layouts/forms/DefaultFormLayout/fields/Checkbox.tsx

```diff
@@ -99,6 +119,34 @@ export const CheckboxField: React.FC<CheckboxProps> = ({ field }) => {
                     <label htmlFor={"checkbox-" + fieldId + option.value}>{option.label}</label>
                 </CheckboxGroup>
             ))}
+            {field.settings.otherOption && (
+                <CheckboxGroup>
+                    <CheckboxButton
+                        name={fieldId}
+                        type="checkbox"
+                        id={"checkbox-" + fieldId + otherOption.value}
+                        checked={checked({ option: otherOption, value })}
+                        onChange={e => {
+                            change({ option: otherOption, value, onChange });
+                            if (e.target.checked && otherInputRef.current) {
+                                otherInputRef.current.disabled = false;
+                                otherInputRef.current.focus();
+                            }
+                        }}
+                    />
+                    <label htmlFor={"checkbox-" + fieldId + otherOption.value}>
+                        {otherOption.label}
+                    </label>
+                    <OtherInput
+                        name={`${fieldId}Other`}
+                        id={`${fieldId}Other`}
+                        ref={otherInputRef}
+                        disabled={!checked({ option: otherOption, value })}
```

Comment:
Couldn't we simply render or not render the `OtherInput`, depending on the `checked` call result? A bit confused why we're using `disabled` here and then hiding the field if it's true (via above CSS).